### ISSUE TITLE
Add inverse kinematics.

### ISSUE THREAD

**Issue by KuehnhammerTobias** [KuehnhammerTobias/ioqw-devel#9]
As shown in #3 ragdoll characters and inverse kinematics (IK), are NOT always the same. Some games have ragdoll characters but no IK. Doom 3 uses IK, while the animation is going on regulary, the feet of a player model, or the wheels of a car, always stay on the floor, even when walking or driving along uneven terrain (IK-Walk). See this:(https://www.iddevnet.com/doom3/walkik.php).  When the object dies the animation is played, forces of the IK influences the animation (showing the 'ragdoll' behaviour of the object.) This is the best approch (imo).

So, what I am trying to do in #3 is to add a physics engine. But is it worth to add 'only' a physics engine, spending years of doing this, only to see that we still don't have IK-Walk?

To me it seems 'better' to fully integrate Doom3 IK. Having Doom3 IK would give use IK-Walk and IK-ragdolls, and Doom3 is GPL3, the same as Spearmint :)

I opened this as a new issue because adding a physics engine won't give us IK-Walk automatically.
If ragdolls are only available by integrating a 3rd Party physics engine, than we have to implement IK-Walk seperatly. This ticket was made for representing this...

In other words, currently I try to implement a physics engine (bullet physics/ODE) into Spearmint, based upon Xreal code. Afaik, Robert Beckebans was the first (and only) one who tried to implement a physics engine into idtech3. The implemention was never finished, it was Java based, and more or less without functionality.
So why not implementing Doom 3 physics after all this? Well, Robert Beckebans implemention is a bit easier to understand than Doom 3 code.

disillusionment...
